Apsim wheat

HamishBrownPFR opened the pull request:
working on #4773

hol430 commented on Models/PMF/Phenology/CAMP/PTQPhyllochron.cs:
```diff
@@ -0,0 +1,60 @@
+        [EventSubscribe("Sowing")]
+        private void OnSowing(object sender, EventArgs e)
+        {
+            lARPTQmodel = phenology.FindChild<LARPTQmodel>("LARPTQmodel");
```
Why not just use a child link by name here?

HamishBrownPFR replied:
Because LARPTQmodel is not in phenology.cs, it is constructed in json.
I know this is a bit messy, we are focusing on getting the model working at the moment and will tidy it up soon

HamishBrownPFR commented:
@hol353 this is OK to merge.  Fixed a very minor bug in structure which has caused a few very small diffs and renamed a variable which has created further diffs.

HamishBrownPFR commented:
@ZhiganZhao  can you disable those SupReports and push again.  Because we are not using camp in the committed version of wheat yet so they are causing errors in the test set which cause the pull request to fail.  OK to leave the change you have made to camp.cs in.

HamishBrownPFR commented:
@ZhiganZhao don't worry, I disabled it.  @hol353 are you happy to merge this?

hol353 commented on Models/PMF/Phenology/CAMP/CAMP.cs:
```diff
@@ -47,11 +74,18 @@ public class ColdUpRegVrn1 : Model, IFunction, IIndexedFunction
         [Description("The temperature above which Vrn1 is down regulated")]
-        public double DeVernalisationTemp { get; set; }
+        [Link(Type = LinkType.Child, ByName = true)]
+        IFunction DeVernalisationTemp = null;
```
Private fields should start with a lower case. See coding guidelines: https://apsimnextgeneration.netlify.app/development/model/2-codingstyle/

hol353 commented on Models/PMF/Phenology/CAMP/CAMP.cs:
```diff
@@ -69,11 +103,11 @@ public double ValueIndexed(double dX)
                 double dHS = camp.dHS / 24;  //divide by 24 to make hourly
-                double UdVrn1 = camp.Params.MaxDVrn1 * Math.Exp(k * dX);
-                if (dX < DeVernalisationTemp)
+                double UdVrn1 = camp.Params.MaxDVrn1 * Math.Exp(k.Value() * dX);
```
Local variables in a method are private and should start with a lower case letter.

hol353 commented on Models/PMF/Phenology/CAMP/CAMP.cs:
```diff
@@ -144,31 +186,46 @@ private double CalcBaseUpRegVrn1(double Tt,double dHS, double BaseDVrn1)
+        /// <param name="IpVrn2"> Initial potential Vrn2 at first experience of Pp > 8 (normally at emergence)</param>
+        /// <param name="DpVrn2">Delta of potential Vrn2 in response to accumulation of LPpHS</param>
+        /// <returns>delta ColdVrn1 representing the additional Vrn1 expression from cold upregulation</returns>
+        private double CalcpVrn2(double LPpHS, double IpVrn2, double DpVrn2)
```
Arguments to a method are also private to that method and so should start with a lower case letter.

hol353 commented on Models/PMF/Phenology/CAMP/CAMP.cs:
```diff
@@ -144,31 +186,46 @@ private double CalcBaseUpRegVrn1(double Tt,double dHS, double BaseDVrn1)
         // Class constants, assumed the same for all cultivars
         /// <summary>Temperature response coefficient for vernalisation</summary>
-        public double k { get { return -0.19; } }
+        public double k { get { return -0.17; } }
```
Public properties should start with an upper case letter.

HamishBrownPFR commented:
@hol353 would you be willing to merge this pull request and I can resolve those style issues in the next.  Camp.cs is only being exercised by the wheat model we have on our branch at the moment so changing them later will not require a converter.  We are quite pressed to make progress with this project at the moment and need to get this pull request merged to do so.

HamishBrownPFR commented:
@hol353 .  The case for the initial character on the variables you point out was used to be consistent with external documentation and prototype code.  Is this acceptable or is it better to change them to meet APSIM requirements.

hol353 commented:
My preference is to have names that are consistent throughout APSIM. If you think it is important to match names with an external model somewhere then I guess that is ok.

HamishBrownPFR commented:
Thanks @hol353  I will keep them as is while we are testing the model but change them to the APSIM convention before we close this issue off .
